kedro-datasets/kedro_datasets/ibis/table_dataset.py

```diff
@@ -0,0 +1,124 @@
+        "overwrite": True,
+    }
+
+    _connections: ClassVar[dict[tuple[tuple[str, str]], BaseBackend]] = {}
+
+    def __init__(  # noqa: PLR0913
+        self,
+        *,
+        filepath: str | None = None,
+        file_format: str | None = None,
+        table_name: str | None = None,
+        connection: dict[str, Any] | None = None,
+        load_args: dict[str, Any] | None = None,
+        save_args: dict[str, Any] | None = None,
+    ) -> None:
+        """Creates a new ``TableDataset`` pointing to a table (or file).
+
+        Args:
+            filepath: Path to a file to register as a table. Most useful
+                for loading data into your data warehouse (for testing).
+            file_format: Specifies the input file format for `filepath`.
+            table_name: The name of the table or view to read or create.
+            connection: Configuration for connecting to an Ibis backend.
+            load_args: Additional arguments passed to the Ibis backend's
+                `read_{file_format}` method.
+            save_args: Additional arguments passed to the Ibis backend's
+                `create_{materialized}` method. By default, ``ir.Table``
+                objects are materialized as views. To save a table using
+                a different materialization strategy, supply a value for
+                `materialized` in `save_args`.
+        """
+        if filepath is None and table_name is None:
+            raise DatasetError(
+                "Must provide at least one of `filepath` or `table_name`."
+            )
+
+        self._filepath = filepath
+        self._file_format = file_format
+        self._table_name = table_name
+        self._connection_config = connection
+
+        # Set load and save arguments, overwriting defaults if provided.
+        self._load_args = deepcopy(self.DEFAULT_LOAD_ARGS)
+        if load_args is not None:
+            self._load_args.update(load_args)
+
+        self._save_args = deepcopy(self.DEFAULT_SAVE_ARGS)
+        if save_args is not None:
+            self._save_args.update(save_args)
+
+        self._materialized = self._save_args.pop("materialized")
+
+    @property
+    def connection(self) -> BaseBackend:
+        cls = type(self)
+        key = tuple(sorted(self._connection_config.items()))
+        if key not in cls._connections:
+            import ibis
+
+            config = deepcopy(self._connection_config)
+            backend = getattr(ibis, config.pop("backend"))
+            cls._connections[key] = backend.connect(**config)
+
+        return cls._connections[key]
+
+    def _load(self) -> ir.Table:
+        if self._filepath is not None:
+            if self._file_format is None:
+                raise NotImplementedError
+
+            reader = getattr(self.connection, f"read_{self._file_format}")
+            return reader(self._filepath, self._table_name, **self._load_args)
+        else:
+            return self.connection.table(self._table_name)
```

Comment:
currently when loading tables thru the `mssql` backend I am doing `conn.table(table_name="XXX", schema="A").` 

an easy solution would be to add the load args into the `conn.table` call and specify the schema as a `load_arg`, but imo this is suboptimal, as the write and save schema should usually (in my experience, my assertion could be wrong) be the same

Reply:
@gforsyth not sure if you have any insight here wrt the upcoming changes in ibis around schema and db hierarchy

Reply:
so starting in 9.0, `schema` will be deprecated as a kwarg.  We will be using the word "database" to refer to a collection of tables and the word "catalog" to refer to a collection of "database".

For specifying only the `database`, you would do:

```
conn.table(table_name="XX", database="A")
```

if the database is located in a catalog (like, say `dbo` or `sys`) you would do either of:

```
conn.table(table_name="XX", database="dbo.A")
```

or

```
conn.table(table_name="XX", database=("dbo", "A"))
```

(`schema` will still work in 9.0, it will just raise a FutureWarning)